lib/services.js

```diff
@@ -286,9 +286,12 @@ const services = {
      *                           delete a specific version of the object
      * @param {Log} log - logger instance
      * @param {function} cb - callback from async.waterfall in objectGet
+     * @param {boolean} deferDeletion - true if the object should not be removed
+     *                                  from the storage, but be returned
+     *                                  instead.
      * @return {undefined}
      */
-    deleteObject(bucketName, objectMD, objectKey, options, log, cb) {
+    deleteObject(bucketName, objectMD, objectKey, options, log, cb, deferDeletion) {
```

Comment:
Since `deferDeletion` refers to **data** locations deletion, it should I think be renamed for clarity, e.g. `deferLocationsDeletion`